autosklearn/data/target_validator.py

```diff
@@ -293,49 +292,83 @@ def is_single_column_target(self) -> bool:
 
     def _check_data(
         self,
-        y: typing.Union[np.ndarray, pd.DataFrame],
+        y: SUPPORTED_TARGET_TYPES,
     ) -> None:
         """
         Perform dimensionality and data type checks on the targets
 
         Parameters
         ----------
-            y: typing.Union[np.ndarray, pd.DataFrame]
+            y: typing.Union[np.ndarray, pd.DataFrame, pd.Series]
                 A set of features whose dimensionality and data type is going to be checked
         """
 
-        if not isinstance(y, (np.ndarray, pd.DataFrame, list, pd.Series)):
+        if not isinstance(
+                y, (np.ndarray, pd.DataFrame, list, pd.Series)) and not scipy.sparse.issparse(y):
             raise ValueError("Auto-sklearn only supports Numpy arrays, Pandas DataFrames,"
-                             " pd.Series and Python Lists as targets, yet, the provided input is"
-                             " of type {}".format(
+                             " pd.Series, sparse data and Python Lists as targets, yet, "
+                             "the provided input is of type {}".format(
                                  type(y)
                              ))
 
+        # Sparse data muss be numerical
+        # Type ignore on attribute because sparse targets have a dtype
+        if scipy.sparse.issparse(y) and not np.issubdtype(y.dtype.type,  # type: ignore[union-attr]
+                                                          np.number):
+            raise ValueError("When providing a sparse matrix as targets, the only supported "
+                             "values are numerical. Please consider using a dense"
+                             " instead."
+                             )
+
         if self.data_type is None:
             self.data_type = type(y)
         if self.data_type != type(y):
-            warnings.warn("Auto-sklearn previously received features of type %s "
-                          "yet the current features have type %s. Changing the dtype "
-                          "of inputs to an estimator might cause problems" % (
-                                str(self.data_type),
-                                str(type(y)),
-                             ),
-                          )
+            self.logger.warning("Auto-sklearn previously received features of type %s "
```

Comment:
I'm only seeing this now, but shouldn't this be `received target(s) of type` instead `received feature(s) of type`?